R/fit.R

```diff
@@ -357,21 +368,33 @@ check_xy_interface <- function(x, y, cl, model) {
         )
       )
 
-  # Determine the `fit()` interface
-  matrix_interface <- !is.null(x) & !is.null(y) && is.matrix(x)
+
+  if (sparse_ok) {
+    matrix_interface <- !is.null(x) & !is.null(y) && (is.matrix(x) | sparse_x)
+  } else {
+    matrix_interface <- !is.null(x) & !is.null(y) && is.matrix(x)
```

Comment:
```suggestion
    matrix_interface <- !is.null(x) && !is.null(y) && (is.matrix(x) | sparse_x)
  } else {
    matrix_interface <- !is.null(x) && !is.null(y) && is.matrix(x)
```

I think this was a typo?

Reply:
yeah. I'll fix that